ios/Classes/AblyFlutterPlugin.m

```diff
@@ -39,6 +39,44 @@ -(nullable AblyFlutter *)ablyWithHandle:(NSNumber *)handle;
     [plugin registerWithCompletionHandler:result];
 };
 
+static FlutterHandler _createRestWithOptions = ^void(AblyFlutterPlugin *const plugin, FlutterMethodCall *const call, const FlutterResult result) {
+    AblyFlutterMessage *const message = call.arguments;
+    LOG(@"message for handle %@", message.handle);
+    AblyFlutter *const ably = [plugin ablyWithHandle:message.handle];
+    // TODO if ably is nil here then an error response, perhaps? or allow Dart side to understand null response?
+    result([ably createRestWithOptions:message.message]);
+};
+
+static FlutterHandler _publishRestMessage = ^void(AblyFlutterPlugin *const plugin, FlutterMethodCall *const call, const FlutterResult result) {
+    AblyFlutterMessage *const message = call.arguments;
+    LOG(@"message for handle %@", message.handle);
+    AblyFlutter *const ably = [plugin ablyWithHandle:message.handle];
+    AblyFlutterMessage *const messageData = message.message;
+    NSMutableDictionary<NSString *, NSObject *>* _dataMap = messageData.message;
+    NSString *channelName = (NSString*)[_dataMap objectForKey:@"channel"];
+    NSString *eventName = (NSString*)[_dataMap objectForKey:@"name"];
+    NSObject *eventData = (NSString*)[_dataMap objectForKey:@"message"];
+    ARTRest *client = [ably getRest:messageData.handle];
+    ARTRestChannel *channel = [client.channels get:channelName];
+
+    //Handling nil cases, to be generalized based on other encounters
+    NSString *nilStr = nil;
+    if([eventName isKindOfClass:[NSNull class]]){ eventName = nilStr; }
+    if([eventData isKindOfClass:[NSNull class]]){ eventData = nilStr; }
+    [channel publish:eventName data:eventData callback:^(ARTErrorInfo *_Nullable error){
```

Comment:
The `_Nullable` here is only of interest to callers, not the implementer, so is superfluous. Also no reason not to make that `error` pointer `const` (swap the `_Nullable` for `const`).

Reply:
Handled in next PR

Reply:
@tiholic I think it would be preferable to have complete PRs, and reduce the sope.  I appreciate there is ongoing work, and you want to get stuff reviewed early, which is good.  However, if the answer is I will fix in the next PR, it means we cannot really review any PRs in full because the feedback is addressed in a separate review, which may or may not be done by the same people.  I think moving forward, we should aim to have all feedback on PRs reviewed and addressed in that PR.  This will ensure we are all happy about code we are reviewing being merged.  WDYT?

Reply:
Yes, agreed @mattheworiordan

Reply:
💯

Reply:
To expand on my agreement with Matt's comment, the ideal is that I can click on the 'Resolve Conversation' button for this comment thread, solely in the context of this PR. Thus once I've seen you push an additional commit to this PR that fixes this issue then I can resolve the comment and therefore we can all visually see progress on this PR towards it being suitable to merge.

Clearly this approach of iterating on this PR comes with the slight downside that your subsequent PR underlying branches (#5 and knock on #6) will need to be rebased on top of the branch underlying this one. But then that's the maintenance overhead of having several chained PRs 'in flight' and is pretty much unavoidable.

Hopefully by sticking to the rules and keeping things clean in terms of branch management and maintenance while we go through the review process will keep the process easy and predictable for you. I'm happy to help if any of that is unclear or you have suggestions for an alternative workflow.